arangod/Graph/Providers/BaseProviderOptions.h

```diff
@@ -56,22 +56,29 @@ struct IndexAccessor {
 
 struct BaseProviderOptions {
  public:
-  BaseProviderOptions(aql::Variable const* tmpVar, std::vector<IndexAccessor> indexInfo);
+  BaseProviderOptions(aql::Variable const* tmpVar, std::vector<IndexAccessor> indexInfo,
+                      std::map<std::string, std::string> const& collectionToShardMap);
 
   aql::Variable const* tmpVar() const;
   std::vector<IndexAccessor> const& indexInformations() const;
 
+  std::map<std::string, std::string> const& collectionToShardMap() const;
+
  private:
   // The temporary Variable used in the Indexes
   aql::Variable const* _temporaryVariable;
   // One entry per collection, ShardTranslation needs
   // to be done by Provider
   std::vector<IndexAccessor> _indexInformation;
+
+  // CollectionName to ShardMap, used if the Traversal is pushed down to DBServer
+  std::map<std::string, std::string> const& _collectionToShardMap;
```

Comment:
Should we count the size in our _resourceMonitor?

Reply:
We only keep the Reference here.
Some other place holds and manages the content.
So if at all the other place needs to keep track.

Also this Map is static, and does not account into the dynamic resources the Resource Manager Tracks.
Will start internal discussion around this.
Thank you for Spotting! But as this is not this PRs responsibility we will merge this.